### ISSUE TITLE
issues not consistently detected inside macros

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#46512]
|  |  |
| --- | --- |
| Bugzilla Link | [47168](https://llvm.org/bz47168) |
| Version | unspecified |
| OS | Windows NT |

## Extended Description 
It appears clang-tidy does not detect issues when they occur in macros consistently. It appears to work with modernize-avoid-c-arrays - see bug 47167 for a related issue - but I couldn't get it to work with performance-move-const-arg or readability-simplify-boolean-expr.

Example (performance-move-const-arg):

```cpp
#include <string>

#define MOVE(x) \
	std::move(x)

int main()
{
	const std::string s;
	const std::string s2 = std::move(s); // warning reported
	const std::string s3 = MOVE(s2); // no warnings
	(void)s3;

	return 0;
}
```

**Comment by vbvictor** [comment]
https://godbolt.org/z/Mq9q73e1Y
